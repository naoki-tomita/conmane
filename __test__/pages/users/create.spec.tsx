import { create, act } from "react-test-renderer";
import Create from "../../../pages/users/create";

describe("create user", () => {
  describe("initial state", () => {
    it("should match with snapshot", () => {
      const renderer = create(<Create />);
      expect(renderer).toMatchSnapshot();
    });
  });

  describe("login control", () => {
    it.todo("should not loginable when id not filled.");
    it.todo("should not loginable when password not filled.");
    it.todo("should loginable when id password filled.");
  });

  describe("exec create user", () => {
    it("should execute create user", async () => {
      const renderer = create(<Create />);
      const idInput = renderer.root
        .find((it) => it.props.label === "id")
        .find((it) => it.type === "input");
      const passwordInput = renderer.root
        .find((it) => it.props.label === "password")
        .find((it) => it.type === "input");
      const doubleCheckPasswordInput = renderer.root
        .find((it) => it.props.label === "double check password")
        .find((it) => it.type === "input");
      const submit = renderer.root.find((it) => it.type === "button");
      global.fetch = (jest.fn(async () => ({
        ok: true,
        json: async () => ({}),
      })) as unknown) as typeof fetch;
      act(() => {
        idInput.props.onChange({ target: { value: "foo" } });
        passwordInput.props.onChange({ target: { value: "bar" } });
        doubleCheckPasswordInput.props.onChange({ target: { value: "bar" } });
      });
      await act(() => submit.props.onClick({ preventDefault: () => "" }));
      expect(global.fetch).toBeCalledWith(
        `http://localhost/api/v1/users/create`,
        {
          method: "POST",
          headers: { "content-type": "application/json" },
          body: JSON.stringify({ loginId: "foo", password: "bar" }),
        }
      );
    });
  });
});
