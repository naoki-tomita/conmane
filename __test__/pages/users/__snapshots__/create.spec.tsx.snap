// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create user initial state should match with snapshot 1`] = `
<form>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <label>
      id
      :
    </label>
    <input
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <label>
      password
      :
    </label>
    <input
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <label>
      double check password
      :
    </label>
    <input
      onChange={[Function]}
      value=""
    />
  </div>
  <button
    onClick={[Function]}
    style={
      Object {
        "width": 120,
      }
    }
  >
    create
  </button>
</form>
`;
